src/ast/irbuilderbpf.cpp

```diff
@@ -282,7 +282,11 @@ CallInst *IRBuilderBPF::createCall(Value *callee,
 {
 #if LLVM_VERSION_MAJOR >= 11
   auto *calleePtrType = cast<PointerType>(callee->getType());
+#if LLVM_VERSION_MAJOR >= 15
+  auto *calleeType = cast<FunctionType>(calleePtrType->getPointerElementType());
```

Comment:
`Type::getPointerElementType` has been in LLVM for a long time, so we should be able to use it for all versions. In fact, here you don't even need to do the above cast to `PointerType`.

This also applies to the rest of changes in this PR.